PhysicsTools/PatUtils/plugins/L1PrefiringWeightProducer.cc

```diff
@@ -0,0 +1,470 @@
+      return 0.;
+
+    prefrate = parametrization1p4To1p6_->Eval(pt);
+    statuncty = parametrization1p4To1p6_->GetParError(2);
+  } else if (std::abs(eta) < 1.8) {
+    if (parametrization1p6To1p8_ == nullptr && !skipwarnings_)
+      edm::LogWarning("L1PrefireWeightProducer")
+          << "Prefiring parametrization not found, setting prefiring rate to 0 " << eta << " " << std::endl;
+    if (parametrization1p6To1p8_ == nullptr)
+      return 0.;
+
+    prefrate = parametrization1p6To1p8_->Eval(pt);
+    statuncty = parametrization1p6To1p8_->GetParError(2);
+  } else if (std::abs(eta) < 2.1) {
+    if (parametrization1p8To2p1_ == nullptr && !skipwarnings_)
+      edm::LogWarning("L1PrefireWeightProducer")
+          << "Prefiring parametrization not found, setting prefiring rate to 0 " << eta << " " << std::endl;
+    if (parametrization1p8To2p1_ == nullptr)
+      return 0.;
+
+    prefrate = parametrization1p8To2p1_->Eval(pt);
+    statuncty = parametrization1p8To2p1_->GetParError(2);
+  } else if (std::abs(eta) < 2.25) {
+    if (parametrization2p1To2p25_ == nullptr && !skipwarnings_)
+      edm::LogWarning("L1PrefireWeightProducer")
+          << "Prefiring parametrization not found, setting prefiring rate to 0 " << eta << " " << std::endl;
+    if (parametrization2p1To2p25_ == nullptr)
+      return 0.;
+
+    prefrate = parametrization2p1To2p25_->Eval(pt);
+    statuncty = parametrization2p1To2p25_->GetParError(2);
+  } else if (std::abs(eta) < 2.4) {
+    if (parametrization2p25To2p4_ == nullptr && !skipwarnings_)
+      edm::LogWarning("L1PrefireWeightProducer")
+          << "Prefiring parametrization not found, setting prefiring rate to 0 " << eta << " " << std::endl;
+    if (parametrization2p25To2p4_ == nullptr)
+      return 0.;
+
+    prefrate = parametrization2p25To2p4_->Eval(pt);
+    statuncty = parametrization2p25To2p4_->GetParError(2);
+  } else
+    return 0.;
+  double systuncty = prefiringRateSystUncMuon_ * prefrate;
+
+  if (fluctuation == up)
+    prefrate = std::min(1., prefrate + sqrt(pow(statuncty, 2) + pow(systuncty, 2)));
+  else if (fluctuation == down)
+    prefrate = std::max(0., prefrate - sqrt(pow(statuncty, 2) + pow(systuncty, 2)));
+  return prefrate;
+}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void L1PrefiringWeightProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("TheMuons", edm::InputTag("slimmedMuons"));
+  desc.add<edm::InputTag>("ThePhotons", edm::InputTag("slimmedPhotons"));
+  desc.add<edm::InputTag>("TheJets", edm::InputTag("slimmedJets"));
+  desc.add<std::string>("L1Maps", "L1PrefiringMaps.root");
+  desc.add<std::string>("L1MuonParametrizations", "L1MuonPrefiringParametriations.root");
+  desc.add<std::string>("DataEraECAL", "2017BtoF");
+  desc.add<std::string>("DataEraMuon", "2016");
+  desc.addUntracked<bool>("UseJetEMPt", false);
```

Comment:
I had asked to make `Untracked` only "SkipWarnings": all other parameters are relevant for the results, and should therefore remain tracked

Reply:
Sorry, misunderstood. I have reversed the change.